src/Functions/PositionImpl.h

```diff
@@ -198,6 +198,48 @@ struct PositionImpl
         const UInt8 * const end = haystack_data.data() + haystack_data.size();
         const UInt8 * pos = begin;
 
+        /// Fastpath when needle is empty
+        if (needle.empty())
+        {
+            if (start_pos == nullptr)
+            {
+                for (auto & x : res)
+                    x = 1;
+                return;
+            }
+
+            ColumnString::Offset prev_offset = 0;
```

Comment:
Please add some comments to the code blocks. (the existing code in l. 243ff does that quite well)